### DISCUSSION TITLE
Powerflow and Weather not displaying

### DISCUSSION THREAD

**Discussion by ViktorJp** [jasonacox/Powerwall-Dashboard#134] [Q&A]
Thanks @jasonacox for your help getting Grafana up to v9!  :)  One last call for help (I hope! :) ), but I can't find a way around this one either.  The powerflow graphic on the dashboard does not show... instead, it just shows some of the code referring to the URL and its port (see screenshot).  I can browse to the host IP where grafana/pwdashboard is running, ie: 192.168.86.101:8675, and it pulls up the powerflow graphic by itself correctly... but when viewing it from within grafana, I get code.  I even tried replacing + "//" + window.location.hostname + ":8675/" with  +'//192.168.86.101:8675/", thinking perhaps it was trying to resolve a hostname that wasn't working right... but no dice.  I don't have any other windows running with the powerflow as you mentioned in another post.  Any ideas?  Thank you!
![powerflowissue](https://user-images.githubusercontent.com/97465574/202710174-1c897452-2e2e-4b34-b0d1-8e8aee599e18.jpg)

And I'm not sure if these are related issues, but I found a couple of other red exclamation marks... the grid status right below the powerflow is showing this:
![gridstatuserror](https://user-images.githubusercontent.com/97465574/202739005-8e9574cc-b4c7-401f-a05c-c3c1a9a96de5.jpg)

And the PW capacity graph is showing this:
![capacityerror](https://user-images.githubusercontent.com/97465574/202739099-4d69f4eb-410e-45f8-b05b-38cf1c129fa9.jpg)

Really makes me concerned that something hasn't been updated correctly on my end... Do you see any reason for me to perhaps try to uninstall everything, and then reinstall, just to see if that would get things back in order?

Last, I got weather411 installed... and I can see weather data by using http://192.168.86.101:8676/temp, but there is no weather data flowing to the dashboard.  Makes me wonder if the link to influxdb is working?  When I do a status, I see these errors:

{"weather411": "0.1.2", "gets": 12, "errors": 6, "timeout": 0, "uri": {"/": 3, "/stats": 2, "/temp": 1}, "ts": 1668791094, "start": 1668787168, "clear": 1668787168, "influxdb": 0, "influxdberrors": 7, "mem": 22712}

**Comment by youzer-name** [comment]
Your first issue looks like this one:  #53

**Reply by ViktorJp** [reply]
Thanks @youzer-name ... unfortunately that setting (GF_PANELS_DISABLE_SANITIZE_HTML=true) was already set that way in the grafana.env file... :(

**Comment by jasonacox** [comment]
This first issue is odd, as if it is not wanting to render HTML which would indicate what @youzer-name mentioned.

Some things to try:

*  Do you see the power flow animation at http://192.168.86.101:8675/example.html ?  That will tell us if pypowerwall is working. Also, most browsers will only allow you to see the animation in one tab so close any that have the animation running.
* Check InfluxdB: `docker logs influxdb` - any obvious errors?
* Ensure InfluxDB has all the CQs set - first make sure you see this file: `influxdb/influxdb.sql` then:

  ```bash
  docker exec --tty influxdb sh -c "influx -import -path=/var/lib/influxdb/influxdb.sql"
  ```
* Re-install Grafana which will require re-adding datasources and importing dashboard.json:
  ```bash
  # Stop and remove grafana
  docker stop grafana
  docker rm grafana
  
  # Get rid of old grafana settings
  mv grafana grafana.old
  mkdir grafana
  
  # Rebuild grafana
  ./compose-dash.sh up -d
  
  # Check startup - will take a while to pull down plugins
  docker logs grafana
  ```
* Go to http://192.168.86.101:9000, login and set up 'InfluxDB' and 'Sun and Moon' datasources 
* Import the dashboard.json dashboard in Grafana

**Reply by ViktorJp** [reply]
OK... unfortunately, after going through this, I'm still seeing the same issue where it's not rendering the powerflow, and those other panels.

1.) I was able to see the iframe perfectly fine when hitting the http://192.168.86.101:8675/example.html... that worked great!

2.) I checked the docker logs for influxdb, and not seeing anything obvious... just regular info-related events.

3.) I ran the docker sql statement.  It did not return any results or confirmation after running.  Is there a way to verify that it imported that right?  NOTE:  One other thing... /var/lib/influxdb is not a valid path.  I don't see influxdb under /var/lib.  I do, however, see it under /Powerwall-Dashboard?   But if I try: docker exec --tty influxdb sh -c "influx -import -path=/Powerwall-Dashboard/influxdb/influxdb.sql"... then it says that "no such file or directory", though it is clearly there. Very confusing. :(

4.) I reran the grafana reinstall, and set it all back up... then imported the .json dashboard... 

So, yeah... unfortunately still seeing that same issue  :(  Please let me know what else I might be able to try?  I truly appreciate your help!

![pwissue](https://user-images.githubusercontent.com/97465574/202832051-34467b66-e6c3-4183-8152-e775ad1c7edd.jpg)

**Reply by jasonacox** [reply]
1) At least that eliminate pypowerwall! 😄 
3) No results is actually good - meaning it ran correctly. The path is confusing because docker does file namespace mapping (it maps `./influxdb` into `/var/lib/influxdb` if you look at powerwall.yml)
4) It still seems like `grafana.env` must be corrupt.  Try this:

```bash
# see if local files are in sync with github
git branch # should say main
git status # should only show untracked files 
cat VERSION # should show 2.6.6

# see if grafana.env is corrupt
diff grafana.env grafana.env.sample # should not see any response

# if those files differ try to copy
cp grafana.env.sample grafana.env
docker stop grafana
docker rm grafana
./compose-dash.sh up -d
```

Finally, just to confirm, can you hover over the "?" on Grafana to make sure it is the right build?

<img width="188" alt="image" src="https://user-images.githubusercontent.com/836718/202834689-23e9870e-cdb8-43cf-8014-9e90e3bfec7f.png">

**Reply by jasonacox** [reply]
Oh!  I just noticed that you mention that the powerwall.yml didn't upgrade and you made the manual edit [here](https://github.com/jasonacox/Powerwall-Dashboard/discussions/132#discussioncomment-4176525).  A lot has changed in that file over time so we need you to get the full file.  That is likely why the grafana.env is not being pulled in. I wonder which other files are not pulling down.

Try this and I suspect you are going to see errors - can you post?

```bash
# pull from repo
git branch #should be main
git switch main # if above is not main
git pull 
```

**Reply by ViktorJp** [reply]
OK... here's the results:

1.) When I typed: git branch - it came back with "main''
2.) When I did: git pull - it threw up a huge list, saying there were 202 commits that I was missing.  I pulled everthing down hoping that would do the trick, restarted everything, and still no dice.
3.) I tried recopying the grafana.env and recomposing, again still no dice.

At that point, I figured perhaps there's just too much going wrong, so I blew the environment away, and started from scratch.

Everything went pretty well, but grafana was complaining about permission issues and wouldn't start, so I did end up setting the GRAFANAUSER to 0:0 in the grafana.env file... 

So after grafana started back up, and getting it configured again, lo and behold... EVERYTHING works.  Even all the panels that were giving me blanks before.  It must have just been hobbling along using old configs/files and with this new dashboard.json... it didn't know what to do.

I really appreciate all your help again @jasonacox ... I guess sometimes it helps just starting over from scratch!  :)

**Reply by jasonacox** [reply]
Thanks for the update @ViktorJp !  I'm glad it is working.  The user you selected means it is running at root, which likely isn't ideal but it will work. If you `echo $UID` on your system, you will find the local user you could use instead. I'm considering updating the defaults to use UID instead of 1000 in `compose.env`.

The upgrade process for this project is still not where I want it to be. It seems to work for minor updates or if someone stays current, but as you experienced, can fail catastrophically. That's an issue I would like to address. 

Did you lose all of your historical data? The folders in the `influxdb` directory could be restored (or a backup if you have one) or you can use this brilliant [tool](https://github.com/jasonacox/Powerwall-Dashboard/tree/main/tools/tesla-history) created by @mcbirse to pull the backup data from the Tesla Cloud into your dashboard.

Good job, again, on getting this to work. 😁

**Reply by ViktorJp** [reply]
Thanks for that info... yeah, I believe it was already configured with 1000:1000, which is my UID, but that didn't seem to matter until I dropped it to 0:0... then it started creating subfolders, installing addins, etc.  I totally get it... it is difficult to keep all these pieces working together in harmony, and on top of that, dealing with all kinds of different flavors and installs out there... it's an amazing feat, and you deserve huge praise for this!

I did lose all my historical data... I didn't know if it would have been tainted or in a state that would have prevented it from working right, or borking up this clean install.  I can't WAIT to try out that tool you mentioned by @mcbirse!  Thanks so much!!

**Reply by ViktorJp** [reply]
Oh that is a beautiful thing, @jasonacox!  Just reimported my 687 days of PW data back into influxdb!  Now I have even more data to display than before!  :)  Thanks again!

**Reply by mcbirse** [reply]
@ViktorJp - Wow that's awesome, great to hear people finding the tool so useful! Just curious how it went pulling down almost 2 years of data...! Did you do that in the one run of the script, and it worked okay without the Tesla servers timing out at all?

**Reply by BJReplay** [reply]
I did five years without an issue.

I did a test run (and it took a while), but seemed to work, and I then tore down and re-started when I'd figured out some issues.

In the end, I was left with about 15 or so 10 minute gaps (mostly 10 minutes, some were longer) for each firmware upgrade.

**Reply by ViktorJp** [reply]
@mcbirse - Yep!  One run... no timeouts.  It took a little more than 1 second per day to pull stuff down.  Nice work!!  :)

**Reply by mcbirse** [reply]
Thanks @BJReplay and @ViktorJp - nice, thank you for the feedback.

I was merely curious, since I only had about a year of data to test with myself.

While I was developing the script, and when I did not have a delay between requests, I had multiple occurrences of Tesla blocking/banning my account for a period of time when requesting a lot of history.

So I added a 1 second default delay between requests, which certainly seems sufficient from your feedback to circumvent this issue. 👍
